universum/modules/structure_handler.py

```diff
@@ -51,23 +51,23 @@ class Block:
     True
     """
 
-    def __init__(self, name: str, parent: 'Block' = None):
-        self.name = name
-        self.status = "Success"
-        self.children = []
+    def __init__(self, name: str, parent: Optional[Block] = None):
+        self.name: str = name
+        self.status: str = "Success"
+        self.children: List[Block] = []
 
-        self._parent = parent
-        self.number = ''
-        if self.parent:
-            self.parent.children.append(self)
+        self._parent: Optional[Block] = parent
+        self.number: str = ''
+        if parent:
```

Comment:
this code never executed before since 'self' doesn't have a 'parent' field - only '_parent'

Reply:
No, there is a 'parent' property.

Reply:
could you please point to a line where it's defined?

Reply:
line 70